solr/core/src/test/org/apache/solr/search/TestDocSet.java

```diff
@@ -556,41 +558,55 @@ public void doFilterTest(IndexReader reader) throws IOException {
     doTestIteratorEqual(da, db);
     ***/
 
-    DocIdSet da;
-    DocIdSet db;
     List<LeafReaderContext> leaves = topLevelContext.leaves();
-
     // first test in-sequence sub readers
     for (LeafReaderContext readerContext : leaves) {
-      da = fa.getDocIdSet(readerContext, null);
-      db = fb.getDocIdSet(readerContext, null);
-
       // there are various ways that disis can be retrieved for each leafReader; they should all be equivalent.
-      doTestIteratorEqual(da.bits(), disiSupplier(da), disiSupplier(db), () -> a.iterator(readerContext), () -> b.iterator(readerContext));
-
+         doTestIteratorEqual(getExpectedBits(a, readerContext), () -> a.iterator(readerContext), () -> b.iterator(readerContext));
       // set b is SortedIntDocSet, so derivatives should not support random-access via Bits
```

Comment:
this comment is now obsolete once you removed the line that followed

Reply:
Good catch! Thanks 👍